src/Microsoft.TemplateEngine.Edge/Settings/TemplateInfo.cs

```diff
@@ -1,145 +1,245 @@
 using System;
 using System.Collections.Generic;
 using Microsoft.TemplateEngine.Abstractions;
-using Microsoft.TemplateEngine.Edge.Settings.TemplateInfoReaders;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 using Microsoft.TemplateEngine.Utils;
 
+#nullable enable
+
 namespace Microsoft.TemplateEngine.Edge.Settings
 {
-    public class TemplateInfo : ITemplateInfo, IShortNameList
+    public partial class TemplateInfo : ITemplateInfo, IShortNameList
     {
-        public static readonly string CurrentVersion = "1.0.0.4";
+        public TemplateInfo(JObject jObject)
+        {
+            MountPointUri = jObject.ToString(nameof(MountPointUri)) ?? throw new ArgumentNullException(nameof(MountPointUri));
+            Author = jObject.ToString(nameof(Author));
+            DefaultName = jObject.ToString(nameof(DefaultName));
+            Description = jObject.ToString(nameof(Description));
+            Identity = jObject.ToString(nameof(Identity)) ?? throw new ArgumentNullException(nameof(Identity));
+            GeneratorId = Guid.Parse(jObject.ToString(nameof(GeneratorId)) ?? throw new ArgumentNullException(nameof(GeneratorId)));
+            GroupIdentity = jObject.ToString(nameof(GroupIdentity));
+            Precedence = jObject.ToInt32(nameof(Precedence));
+            Name = jObject.ToString(nameof(Name)) ?? throw new ArgumentNullException(nameof(Name));
+            ConfigPlace = jObject.ToString(nameof(ConfigPlace)) ?? throw new ArgumentNullException(nameof(ConfigPlace));
+            LocaleConfigPlace = jObject.ToString(nameof(LocaleConfigPlace));
+            HostConfigPlace = jObject.ToString(nameof(HostConfigPlace));
+            ThirdPartyNotices = jObject.ToString(nameof(ThirdPartyNotices));
+            HasScriptRunningPostActions = jObject.ToBool(nameof(HasScriptRunningPostActions));
+
+            ShortNameList = JTokenStringOrArrayToCollection(jObject.Get<JToken>(nameof(ShortNameList)), new string[0]);
+            if (ShortNameList.Count == 0)
+            {
+                throw new Exception($"Cache is missing {ShortNameList}");
```

Comment:
For the future: maybe we can have our own set of exception types instead of using `System.Exception`.